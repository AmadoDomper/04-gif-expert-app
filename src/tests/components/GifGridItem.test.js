import React from "react";
import {shallow} from "enzyme";
import GifGridItem from "../../Components/GifGridItem";


describe('Testing in <GifGridItem />', () => {

    const title = 'A tittle';
    const url = 'https://localhost/something.jpg'

    let wrapper = shallow(<GifGridItem title={ title } url={ url } />)

    test('should show <GifGridItem /> correctly', () => {
        
        expect(wrapper).toMatchSnapshot();
    })

    test('should have a paragraph with the title', () => {
        const titleText = wrapper.find('p').text().trim();

        expect(titleText).toBe(title);
    })
    
    test('should have src and alt', () => {
        
        const imgSrc = wrapper.find('img').prop("src").trim();
        const imgAlt = wrapper.find('img').prop("alt").trim();

        expect(imgSrc).toBe(url);
        expect(imgAlt).toBe(title);
        
    })
    
    test('should have animate__fadeIn', () => {
        const div = wrapper.find('div');

        expect(div.hasClass('animate__fadeIn')).toBe(true);
    })
    
    
    
})
