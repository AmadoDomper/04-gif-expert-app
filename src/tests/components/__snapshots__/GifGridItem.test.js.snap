// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <GifGridItem /> should show <GifGridItem /> correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-2s"
>
  <img
    alt="A tittle"
    src="https://localhost/something.jpg"
  />
  <p>
     
    A tittle
  </p>
</div>
`;
